public/chart.js

```diff
@@ -161,18 +227,39 @@ function characterChart(svg, dataURL) {
         recalc();
     }
 
+    function zoomed() {
+        // Workaround because .xExtent() for the zoom behaviour is not part of the current official d3js release
+        if (x.domain()[0] < fullXdomain[0]) {
+            zoom.translate([zoom.translate()[0] - x(fullXdomain[0]) + x.range()[0], 0]);
+        } else if (x.domain()[1] > fullXdomain[1]) {
+            zoom.translate([zoom.translate()[0] - x(fullXdomain[1]) + x.range()[1], 0]);
+        }
+
+        recalc();
+    }
+
     function recalc() {
         // areas
         pos.attr("d", calcAreaPos);
         neg.attr("d", calcAreaNeg);
 
+        // episode labels
+        d3.selectAll(".episode")
+            .attr("x", function (d) {
+                return x(d.date);
+            });
+
         // axis
         xLabel.call(xAxis);
         yLabel.call(yAxis);
+        eLabel.call(eAxis);
     }
 
-    // request csv data and fill chart
-    d3.csv(dataURL, convertCSV, fillChart);
+    // request csv twitter data and fill chart
+    d3.csv(dataURL, convertTwitterCSV, fillChart);
+
+    // request csv episodes data and create episode labels
+    d3.csv(episodesURL, convertEpisodesCSV, episodeLabels)
 
     this.resize = render;
```

Comment:
What if we do just that in chart.html?

`var` chart = new characterChart(d3.select("#chart"), "/csv/{{slug}}.csv");

Then the this in characterChart actually references the function